test-utils/test-framework/src/framework.rs

```diff
@@ -0,0 +1,116 @@
+use actix_rt;
+use anyhow;
+use std::future::Future;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use crate::YagnaMock;
+
+pub use macros::prepare_test_dir;
+pub use ya_framework_macro::framework_test;
+
+#[derive(Clone)]
+pub struct YagnaFramework {
+    inner: Arc<Mutex<YagnaNetworkImpl>>,
+    test_dir: PathBuf,
+    test_name: String,
+}
+
+/// Entities that require tear down after test is finished.
+/// This allows us to kill yagna process even in case of panic.
+#[derive(Clone)]
+pub struct YagnaNetworkImpl {
+    nodes: Vec<YagnaMock>,
+}
+
+pub mod macros {
+    #[macro_export]
+    macro_rules! prepare_test_dir {
+        () => {
+            // CARGO_TARGET_TMPDIR is available in compile time only in binary modules, so we can't
+            // use it in this library. Thanks to using macro it will be resolved in final code not here
+            // and it will work.
+            ya_test_framework::framework::prepare_test_dir_(env!("CARGO_TARGET_TMPDIR"))
+        };
+    }
+    pub use prepare_test_dir;
+    pub use serial_test;
+}
+
+pub fn prepare_test_dir_(dir: &str) -> PathBuf {
+    let test_dir = PathBuf::from(&dir);
+
+    #[allow(unused_must_use)]
+    {
+        std::fs::remove_dir_all(&test_dir); // ignores error if does not exist
```

Comment:
[Question]
1. Will it remove dirs also after some tests failed? If yes, than it will become impossible to debug such test as there won't be any logs from yagna.
2. I wonder do we want to reuse existing test directories or make another ones with folder of different timestamp (as in Goth or other e2e testing frameworks)?

I believe reusing / removing directories after each run is enough for UT or IT but for e2e ones it will hurt us sooner rather than later.

Reply:
Directories will remain after test, but will be removed on next run.
Someone running tests can consciously choose to save directory by just changing it's name.

Do you think it's enough?